specification/data-semantic-conventions.md

```diff
@@ -279,33 +284,67 @@ but due to `http.scheme`, `http.host` and `http.target` being set, it would be r
-| `db.instance`  | Database instance name. E.g., In Java, if the jdbc.url=`"jdbc:mysql://db.example.com:3306/customers"`, the instance name is `"customers"`. | Yes     |
-| `db.statement` | A database statement for the given database type. Note, that the value may be sanitized to exclude sensitive information. E.g., for `db.type="sql"`, `"SELECT * FROM wuser_table"`; for `db.type="redis"`, `"SET mykey 'WuValue'"`. | Yes       |
-| `db.url` | The connection string used to connect to the database | Yes       |
-| `db.resultcount` | An integer specifying the number of results returned. | No       |
-| `db.roundtripcount` | An integer specifying the number of network roundtrips while executing the request. | No       |
+| `db.type` | The lower-case database category, e.g. `"sql"`, `"cassandra"`, `"hbase"`, or `"redis"`. | Yes       |
+| `db.tech` | Name of the database technology used, e.g. "SQL Server" or "MySQL". | No     |
+| `db.url` | The connection string used to connect to the database | No       |
 | `db.user`      | Username for accessing database. E.g., `"readonly_user"` or `"reporting_user"` | No        |
+| `db.portname` | A name that identifies the `net.peer.port` used to connect. This property is intended to be used for [MS SQL Server instances][] but can be used for any database system that provides a name to port mapping. | See below        |
+
+[MS SQL Server instances]: https://docs.microsoft.com/en-us/sql/connect/jdbc/building-the-connection-url?view=sql-server-ver15
+
+Additionally at least one of `net.peer.name` or `net.peer.ip` from the [network attributes][] is required and `net.peer.port` is recommended.
+If using a non-standard port for the `db.tech`, at least one of `net.peer.port` or `db.portname` is required.
+
+### Call-level attributes
+
+These attributes may be different for each operation performed, even if it uses the same connection
+although usually only one `db.name` will be used per connection.
+
+| Attribute name | Notes and examples                                           | Required? |
+| :------------- | :----------------------------------------------------------- | --------- |
+| `db.name`  | Database name. E.g., In Java, if the jdbc.url=`"jdbc:mysql://db.example.com:3306/customers"`, the name is `"customers"`. Note that this attribute was previously called `db.instance` but has nothing to do with [MS SQL Server instances][]. For commands that switch the database, this should be set to the target database (even if the command fails). | Yes (if applicable) |
+| `db.statement` | A database statement for the given database type. Note that the value may be sanitized to exclude sensitive information. E.g., for `db.type="sql"`, `"SELECT * FROM wuser_table"`; for `db.type="redis"`, `"SET mykey 'WuValue'"`. | Yes (if applicable)       |
```

Comment:
I would perfer not having a generic `statement`representation:
* `db.sql` should explicitly contain a query that can be parsed, normalized and sanitized as SQL.
* `db.<tech_namespace>.query` should contain the specific NoSQL dialect query which can be normalized in case it is well known, or discarded to not "leak" sensitive data.

Reply:
If going that route, I'd prefer `db.sql.statement`, and `db.<other tech>.statement`.

Reply:
Make sense

Reply:
Should be add also some limitations regarding size or how to encode binary data of databases where queries are no strings?

Reply:
I think we only want to send the statement if it is already text. Others dbs should rather use `db.operation` (which not as "exact" as db.statement). I think I should/did add DB-specific notes below.